model/Metadata/Factory/ClassPathFactory.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoAdvancedSearch\model\Metadata\Factory;
+
+use core_kernel_classes_Class;
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoAdvancedSearch\model\Metadata\Service\MetadataResultSearcher;
+
+class ClassPathFactory extends ConfigurableService
+{
+    use OntologyAwareTrait;
+
+    public function create(core_kernel_classes_Class  $class): array
+    {
+        $path = [$class->getUri()];
+
+        foreach ($class->getParentClasses(true) as $parentClass) {
+            if ($this->isRootClass($class)) {
+                break;
+            }
+
+            if ($this->isRootClass($parentClass)) {
+                $path[] = $parentClass->getUri();
+
+                break;
+            }
+
+            $path[] = $parentClass->getUri();
+        }
+
+        return $path;
```

Comment:
Do we need to check root class on every iteration?

```suggestion
        $path = [$class->getUri()];
        
        if ($this->isRootClass($class)) {
            return $path;
        }

        foreach ($class->getParentClasses(true) as $parentClass) {
            if ($this->isRootClass($parentClass)) {
                $path[] = $parentClass->getUri();

                break;
            }

            $path[] = $parentClass->getUri();
        }

        return $path;
```

Reply:
Good point! Will improve, thanks!

Reply:
Improved 👍